### DISCUSSION TITLE
How to output traceid/segmentid/... into the logs?

### DISCUSSION THREAD

**Discussion by jadezhong** [apache/skywalking#7605] [Q&A] [answered]
How to output traceid/segmentid/... into the logs? such as the log printed with log4j/logback, just like Zipkin does.

**Comment by wu-sheng** [comment, marked as answer]
How about reading docs? https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-log4j-1.x/

**Reply by jadezhong** [reply]
yes, that's what i needed, thx!

**Reply by jadezhong** [reply]
where can i find this section of doc under 8.4.0?

**Reply by wu-sheng** [reply]
https://skywalking.apache.org/docs/#SkyWalking

**Reply by jadezhong** [reply]
can't find the chapter talking about output traceId/context into logs under log4j/logback.

**Reply by wu-sheng** [reply]
We never said this feature is in every previous release. Locating the correct version is your own responsibility.

**Comment by jadezhong** [comment]
sovled. refer to https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/application-toolkit-log4j-1.x/, traceId can be output into logs using logback & logstash under v8.4.0, skywalking context can't.
